### PR TITLE
Adding metric guidelines to the repository

### PR DESCRIPTION
**Pull request by ldematte**
Converting the latest version of the guidelines from gdocs to markdown, and adding them to :modules:apm so they can (later) referenced by the validation code.

### CODE REVIEW

**Comment by ldematte** [comment]
@elasticmachine update branch

File: modules/apm/NAMING.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+Examples :
+* <code>es.process.jvm.collection.<strong>time</strong></code> instead of <code>es.process.jvm.collection.<strong>seconds</strong></code>.
+* <code>es.process.mem.virtual.<strong>size</strong></code>, <code>es.indices.storage.<strong>size</strong></code> (instead of <code>es.process.mem.virtual.<strong>bytes</strong></code>, <code>es.indices.storage.<strong>bytes</strong></code>)
+* In case `size` has a known upper limit, consider using `usage` (e.g.: <code>es.process.jvm.heap.<strong>usage</strong></code> when there is a <code>es.process.jvm.heap.<strong>limit</strong></code>)
+* <code>es.indices.storage.write.<strong>io</strong></code>, instead of <code>es.indices.storage.write.<strong>bytes_per_sec</strong></code>
+* These can all be composed with the suffixes below, e.g. <code>es.process.jvm.collection.<strong>time.total</strong></code>, <code>es.indices.storage.write.<strong>total</strong></code> to represent the monotonic sum of time spent in GC and the total number of bytes written to indices respectively.
+
+**Pluralization** and **suffixes**:
+
```
Comment (rjernst) [inline modules/apm/NAMING.md:52]: nit: Unnecessary extra newlines?

File: modules/apm/NAMING.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+### Attributes
+
+Attribute names should follow the same rules. In particular rules about:
+* elements and separators
+* hierarchy/namespaces
+* units
+* pluralization (when an attribute represents a measurement)
+
+apply to attributes too.
```
Comment (rjernst) [inline modules/apm/NAMING.md:71]: consider moving this up to the line before the list, eg:

```
In particular, these rules apply to attributes too:
```

**Review by rjernst — APPROVED** [review]
This looks good. I do wonder if this is the right place for the doc though. It is buried inside a module. I find it unlikely anyone will find it. Perhaps it should be inline within CONTRIBUTING.md? Or have a new DEVELOPMENT.md or something similar, at the root. In any case, it can be moved as a followup if there is contention on where it should exist.

**Comment by ldematte** [comment]
Thanks Ryan. 
@pgomulka is planning to add a METRICS.md too; when that is in we can review if we want keep them here, merge them, and/or add a link in CONTRIBUTING.md to these files. Wdyt?

**Comment by rjernst** [comment]
Sounds like a good plan.

**Pull request merged**